core/src/main/java/org/apache/calcite/tools/RelBuilder.java

```diff
@@ -2896,6 +2896,9 @@ private static List<RexLiteral> convertLiteralTypes(RelOptCluster cluster,
       RelDataType rowType, List<RexLiteral> values) {
     assert values.size() == rowType.getFieldCount();
     RexExecutor executor = cluster.getPlanner().getExecutor();
+    if (executor == null) {
+      executor = RexUtil.EXECUTOR;
+    }
     if (executor == null) {
```

Comment:
Need to delete.

Reply:
I don't understand this comment.

Reply:
I think @NobiGo means that the currently the patch code is:
```
if (executor == null) {
  executor = RexUtil.EXECUTOR;
}
if (executor == null) {
  return values;
}
```
The second `if` could be removed.

Reply:
I'm not sure RexSimplify#simlify is necessarily needed here. if not, the user should be allowed to set Executor to NULL. Instead, I think
`RexExecutor rexImpl = Util.first(cluster.getPlanner().getExecutor(), RexUtil.EXECUTOR);`
is good

Reply:
If you don't simplify the expression won't be a RexLiteral

Reply:
But turns out that the RelBuilder already contains a simplifier